### PR TITLE
[Reviewer AJH] Pass timeout override to HTTPClient

### PR DESCRIPTION
**Pull request by steelyvoid**
Companion PR to https://github.com/Metaswitch/cpp-common/pull/663

### CODE REVIEW

**Comment by AlexHockey** [comment]
Presumably this change is just to get coverage of your new code. Is there any way to regressibly test that your change is actually working correctly?

**Comment by steelyvoid** [comment]
Thanks for the pointers on the curl interposer, Alex.  Can you double check that this test is now good?  I've also pushed the curl interposer changes to Metaswitch/cpp-common#663.

File: src/ut/httpclient_test.cpp [hunk 1]
```diff
@@ -150,7 +154,10 @@ TEST_F(HttpClientTest, SasOmitBody)
 
   _private_http->send_post("http://cyrus/blah/blah/blah", headers, test_body, 0, "gandalf");
 
-  req_event = mock_sas_find_event(SASEvent::TX_HTTP_REQ);
+    // This client will be using the override timeout (1000)
+  Request& req = fakecurl_requests["http://cyrus:80/blah/blah/blah"];
+  EXPECT_EQ(1000, req._timeout_ms);req_event = mock_sas_find_event(SASEvent::TX_HTTP_REQ);
```
Comment (AlexHockey) [inline src/ut/httpclient_test.cpp]: Looks like this second statement should be on a new line?

**Review by AlexHockey — COMMENTED** [review]
One minor comment inline.

**Pull request merged**